ec_deployment: Add config block to elasticsearch (#65)
elastic/terraform-provider-ec: docs/resources/ec_deployment.md

> The following arguments are supported:

Reviewer: Several of these are in https://github.com/elastic/terraform-provider-ec/blob/master/docs/datasources/ec_deployment.md as well, might be good to keep them consistent? I'm not too attached to what I wrote on there, so if you think maybe we should change some of them that's fine, but I think they should both say the same thing if the field is the same :)

> * `region` - (Required) ESS region where to create the deployment, for ECE environments "ece-region" must be set.

Reviewer: ```suggestion
* `region` - (Required) ESS region where to create the deployment. For ECE environments, "ece-region" must be set.
```

* `deployment_template_id` - (Required) Deployment Template identifier to create the deployment from.
* `version` - (Required) Elastic Stack version to use for all of the deployment resources.

> * `name` - (Optional) name for the deployment.

Reviewer: ```suggestion
* `name` - (Optional) Name for the deployment.
```

> * `request_id` - (Optional) request_id to set on the create operation, only use when previous create attempts return with an error and a request_id is returned as part of the error.

Reviewer: ```suggestion
* `request_id` - (Optional) Request ID to set on the create operation. Only use when previous create attempts return with an error and a request_id is returned as part of the error.
```

* `elasticsearch` (Required) Elasticsearch cluster definition, can only be specified once.
* `kibana` (Optional) Kibana instance definition, can only be specified once.
* `apm` (Optional) APM instance definition, can only be specified once.

> * `enterprise_search` (Optional) APM server definition, can only be specified once.

Reviewer: ```suggestion
* `enterprise_search` (Optional) Enterprise Search server definition, can only be specified once.
```

> * `traffic_filter` (Optional) traffic_filter block, which contains a list of traffic filter rule identifiers.

Reviewer: ```suggestion
* `traffic_filter` (Optional) Traffic filter block, which contains a list of traffic filter rule identifiers.
```

### Resources

In order to be able to create a valid deployment at least one resource type must be specified, below are the supported resources.

#### Elasticsearch

The required `elasticsearch` block supports the following:

> * `topology` - (Required) topology element which must be set once but can be set multiple times to compose complex topologies.

Reviewer: ```suggestion
* `topology` - (Required) Topology element which must be set once, but can be set multiple times to compose complex topologies.
```

* `ref_id` - (Optional) ref_id to set on the Elasticsearch resource (Defaults to `main-elasticsearch`).
* `config` (Optional) Elasticsearch settings which will be applied to all topologies unless overridden on the topology element. 

##### Topology

The required `elasticsearch.topology` block supports the following:

* `instance_configuration_id` - (Required) Instance Configuration ID from the deployment template.
* `memory_per_node` - (Optional) Amount of memory per node in the "<size in GB>g" notation (Defaults to `4g`).

> * `zone_count` - (Optional) Number of zones that the Elasticsearch cluster will span, this is used to set HA (Defaults to `1`).

Reviewer: ```suggestion
* `zone_count` - (Optional) Number of zones that the Elasticsearch cluster will span. This is used to set HA (Defaults to `1`).
```

* `node_type_data` - (Optional) Node type (data) for the Elasticsearch Topology element (Defaults to `true`) 
* `node_type_master` - (Optional) Node type (master) for the Elasticsearch Topology element (Defaults to `true`)
* `node_type_ingest` - (Optional) Node type (ingest) for the Elasticsearch Topology element (Defaults to `true`)

> * `node_type_ml` - (Optional) Node type (Machine Learning) for the Elasticsearch Topology element (Defaults to `false`).

Reviewer: ```suggestion
* `node_type_ml` - (Optional) Node type (machine learning) for the Elasticsearch Topology element (Defaults to `false`).
```

* `config` (Optional) Elasticsearch settings which will be applied at the topology level. 

##### Config

The optional `elasticsearch.config` and `elasticsearch.topology.config` blocks support the following:

> * `plugins` - (Optional) List of Elasticsearch supported plugins, which varies version to version, check the Stack Pack version check which plugins are supported for each version.

Reviewer: ```suggestion
* `plugins` - (Optional) List of Elasticsearch supported plugins, which vary from version to version. Check the Stack Pack version to see which plugins are supported for each version.
```
